todo/src/reducers/index.js

```diff
@@ -0,0 +1,61 @@
+import { ADDTODO, TOGGLETODO, CLEAR } from '../actions';
+
+const initialState = {
+  todos: [{
+    task: 'Organize Garage',
+    id: 1528817077286,
+    completed: false
+  },
+  {
+    task: 'Bake Cookies',
+    id: 1528817084358,
+    completed: false
+  },
+  {
+    task: 'Create TodoList',
+    id: 1528817084395,
+    completed: false
+  },
+  {
+    task: 'Learn Redux',
+    id: 1528817084975,
+    completed: false
+  }]
+}
+
+export const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case ADDTODO:
+    const newTodo = {
+      task: action.payload,
+      id: Date.now(),
+      completed: false
+    }
+      return {
+        ...state,
+        todos: [...state.todos, newTodo]
+      }
+
+    case TOGGLETODO:
+      return {
+        ...state,
+        todos: state.todos.map(todo => {
+          if (todo.id === action.payload) {
+            return {
+              ...todo,
+              completed: !state.completed
```

Comment:
there's an interesting error here -- your app lets you click an item and it toggles once to complete, but you can't turn it back again. that's because of this line here, you should be referencing the current `completed` prop on the actual _todo_ instead of the overall state: `completed: !todo.completed`. As it stands there is no `completed` prop on state, so `state.completed` is a falsey statement, so `!state.completed` is a truthy statement